api/query.py

```diff
@@ -64,8 +64,20 @@ def post(self):
                             results = col_selected.get_ranges([range1])
                             res = json.dumps(results, default=json_serializer)
                             data = json.loads(res)
+                            len_serie = len(data)
                             datapoints = []
-                            for i in xrange(0, len(data)):
+
+                            step = 1
```

Comment:
https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#basic-slicing-and-indexing

Reply:
@edouard Would this case be clearer / more readable using slicing ?
Does the below can cause performances issues?
``` python
for i in xrange(0, len_serie, step):
    datapoints.append([int(data[i][1]), int(data[i][0])])
```

Reply:
The problem is how you compute the step.

Reply:
If you know you want 20,000 points, then steps is (points_count / 20,000)

Reply:
I agree with that, but you've to check if there's at least 20000 points before doing the computation? It depends actually.